File: backend/src/main/java/com/ecommerce/backend/service/CategoryService.java
package com.ecommerce.backend.service;

import com.ecommerce.backend.dto.CategoryDTO;
import com.ecommerce.backend.dto.view.CategoryView;
import com.ecommerce.backend.model.Category;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

public interface CategoryService {

    List<Category> getAllCategories();

    Category getCategoryById(Long id);

    Category createCategory(CategoryDTO categoryDTO); // Changed parameter to CategoryDTO

    Category updateCategory(Long id, CategoryDTO categoryDTO); // Changed parameter to CategoryDTO

    void deleteCategory(Long id);

    String uploadCategoryImage(MultipartFile imageFile); // Added for image upload

    Page<Category> searchCategories(String name, Pageable pageable);

    List<CategoryView> searchByName(String name);
}

File: frontend_client_mobile/lib/services/category_service.dart
import 'package:dio/dio.dart';
import 'package:frontend_client_mobile/models/category.dart';
import 'package:frontend_client_mobile/services/api/api_client.dart';
import 'package:frontend_client_mobile/services/api/category_api_service.dart';

class CategoryService {
  final CategoryApiService _apiService = ApiClient.getCategoryApiService();

  Future<List<Category>> getCategories({String? name}) async {
    return await _apiService.getCategories(name);
  }

  Future<Category> createCategory(Category category) async {
    return await _apiService.createCategory(category);
  }

  Future<Category> updateCategory(int id, Category category) async {
    return await _apiService.updateCategory(id, category);
  }

  Future<void> deleteCategory(int id) async {
    return await _apiService.deleteCategory(id);
  }

  Future<String> uploadCategoryImage(MultipartFile imageFile) async {
    return await _apiService.uploadCategoryImage(imageFile);
  }
}
